crates/ruma-events/src/room/message/url_preview.rs

```diff
@@ -0,0 +1,320 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    file::EncryptedContent,
+    room::{OwnedMxcUri, UInt},
+};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+/// Encrypted Image Preview info
+pub struct EncryptedPreviewImage {
+    /// The target URI
+    pub url: OwnedMxcUri,
+    /// Encrypted PreviewImage data information
+    #[serde(flatten)]
+    pub encrypted_image: Box<EncryptedContent>,
+}
+
+/// UrlPreview Metadata about an image.
+/// modelled after [OpenGraph Image Properties](https://ogp.me/#structured)
```

Comment:
nit: This should be in another paragraph:

```suggestion
///
/// Modelled after [OpenGraph Image Properties](https://ogp.me/#structured).
```